common/src/main/java/com/wynntils/mc/mixin/ClientPacketListenerMixin.java

```diff
@@ -34,10 +35,15 @@
 
 @Mixin(ClientPacketListener.class)
 public abstract class ClientPacketListenerMixin {
+    private static boolean isRenderThread() {
+        return (McUtils.mc().isSameThread());
+    }
+
     @Inject(
             method = "handleCommands(Lnet/minecraft/network/protocol/game/ClientboundCommandsPacket;)V",
             at = @At("HEAD"))
     private void handleCommandsPre(ClientboundCommandsPacket packet, CallbackInfo ci) {
+        if (!isRenderThread()) return;
```

Comment:
I would've expected this more on the worker thread then the render but 🤷

Reply:
What do you mean by "worker" thread? The "Render" thread is the main game thread. I'd had called it "Main" but Minecraft calls it "Render Thread" so I stick with that.

Reply:
Ah, now I get it